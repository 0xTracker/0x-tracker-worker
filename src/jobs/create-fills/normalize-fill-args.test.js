const normalizeFillArgs = require('./normalize-fill-args');

describe('normalizeFillArgs', () => {
  it('should normalize args for V2 event with recognized assets', () => {
    const event = {
      _id: '5bb1f16862f9ca0004c7cfb7',
      blockNumber: 6363524,
      data: {
        address: '0x4f833a24e1f95d70f028921e27040ca56e09ab0b',
        blockHash:
          '0xdb670f0c4934f24a62928ab9c676d4e24b7d7f0d1467c696c82c1789571da3af',
        blockNumber: 6363524,
        data:
          '0x000000000000000000000000f3f694cbe3a1042a886ce923fab00ca649e6a89b000000000000000000000000f3f694cbe3a1042a886ce923fab00ca649e6a89b000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000005af3107a4000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000018000000000000000000000000000000000000000000000000000000000000000440257179200000000000000000000000006012c8cf97bead5deae237070f9587f8e7a266d00000000000000000000000000000000000000000000000000000000000d29fe000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000024f47261b0000000000000000000000000e41d2489571d322189246dafa5ebde1f4699f49800000000000000000000000000000000000000000000000000000000',
        logIndex: 91,
        removed: false,
        topics: [
          '0x0bcc4c97732e47d9946f229edb95f5b6323f601300e4690de719993f3c371129',
          '0x0000000000000000000000005b50fb8281cb679a3102d3de22dd05a1ad895e8f',
          '0x0000000000000000000000000000000000000000000000000000000000000000',
          '0xb0d28d8c36f40322408b3ff4ac41d2bf6ac6c834633763d3b42831d50e30b455',
        ],
        transactionHash:
          '0x266b911f58bc4138616d1e37dcd9829a5f502322b09f99d48d4610e25d6a1e0d',
        transactionIndex: 107,
        event: 'Fill',
        args: {
          makerAddress: '0x5b50fb8281cb679a3102d3de22dd05a1ad895e8f',
          feeRecipientAddress: '0x0000000000000000000000000000000000000000',
          takerAddress: '0xf3f694cbe3a1042a886ce923fab00ca649e6a89b',
          senderAddress: '0xf3f694cbe3a1042a886ce923fab00ca649e6a89b',
          makerAssetFilledAmount: 1,
          takerAssetFilledAmount: 100000000000000,
          makerFeePaid: 0,
          takerFeePaid: 0,
          orderHash:
            '0xb0d28d8c36f40322408b3ff4ac41d2bf6ac6c834633763d3b42831d50e30b455',
          makerAssetData:
            '0x0257179200000000000000000000000006012c8cf97bead5deae237070f9587f8e7a266d00000000000000000000000000000000000000000000000000000000000d29fe',
          takerAssetData:
            '0xf47261b0000000000000000000000000e41d2489571d322189246dafa5ebde1f4699f498',
        },
      },
      logIndex: 91,
      protocolVersion: 2,
      transactionHash:
        '0x266b911f58bc4138616d1e37dcd9829a5f502322b09f99d48d4610e25d6a1e0d',
      type: 'Fill',
      fillCreated: true,
    };

    const normalized = normalizeFillArgs(event.data.args, 2);

    expect(normalized).toMatchSnapshot();
  });
});
