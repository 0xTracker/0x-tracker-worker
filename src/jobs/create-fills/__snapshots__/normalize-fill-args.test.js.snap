// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeFillArgs should normalize args for V1 event 1`] = `
Object {
  "assets": Array [
    Object {
      "actor": 0,
      "amount": 1292498732163907300,
      "tokenAddress": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
      "tokenResolved": false,
      "tokenType": 0,
    },
    Object {
      "actor": 1,
      "amount": 4.2620995342647e+23,
      "tokenAddress": "0x58a4884182d9e835597f405e5f258290e46ae7c2",
      "tokenResolved": false,
      "tokenType": 0,
    },
  ],
  "feeRecipient": "0xe269e891a2ec8585a378882ffa531141205e92e9",
  "filledMakerTokenAmount": 1292498732163907300,
  "filledTakerTokenAmount": 4.2620995342647e+23,
  "maker": "0x0e7c68992fe035f4ed9607866483cf6ae7db6b71",
  "makerToken": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
  "orderHash": "0x9af44f4536831684cacda6dbba6af9974f0b9e8b2fc271f4fa8ff18c6fd51e12",
  "paidMakerFee": 0,
  "paidTakerFee": 0,
  "taker": "0xe269e891a2ec8585a378882ffa531141205e92e9",
  "takerToken": "0x58a4884182d9e835597f405e5f258290e46ae7c2",
}
`;

exports[`normalizeFillArgs should normalize args for V2 event with recognized assets 1`] = `
Object {
  "assets": Array [
    Object {
      "actor": 0,
      "amount": 1,
      "tokenAddress": "0x06012c8cf97bead5deae237070f9587f8e7a266d",
      "tokenId": 862718,
      "tokenResolved": false,
      "tokenType": 1,
    },
    Object {
      "actor": 1,
      "amount": 100000000000000,
      "tokenAddress": "0xe41d2489571d322189246dafa5ebde1f4699f498",
      "tokenId": undefined,
      "tokenResolved": false,
      "tokenType": 0,
    },
  ],
  "feeRecipient": "0x0000000000000000000000000000000000000000",
  "filledMakerTokenAmount": 1,
  "filledTakerTokenAmount": 100000000000000,
  "maker": "0x5b50fb8281cb679a3102d3de22dd05a1ad895e8f",
  "makerAsset": Object {
    "assetProxyId": "0x02571792",
    "tokenAddress": "0x06012c8cf97bead5deae237070f9587f8e7a266d",
    "tokenId": 862718,
  },
  "makerToken": "0x06012c8cf97bead5deae237070f9587f8e7a266d",
  "orderHash": "0xb0d28d8c36f40322408b3ff4ac41d2bf6ac6c834633763d3b42831d50e30b455",
  "paidMakerFee": 0,
  "paidTakerFee": 0,
  "senderAddress": "0xf3f694cbe3a1042a886ce923fab00ca649e6a89b",
  "taker": "0xf3f694cbe3a1042a886ce923fab00ca649e6a89b",
  "takerAsset": Object {
    "assetProxyId": "0xf47261b0",
    "tokenAddress": "0xe41d2489571d322189246dafa5ebde1f4699f498",
  },
  "takerToken": "0xe41d2489571d322189246dafa5ebde1f4699f498",
}
`;
